### PR TITLE
chore(types): add top level Typescript types

### LINKED ISSUE #461: Add typescript definitions for the viewer
To be used:

- https://github.com/InsightSoftwareConsortium/itk-viewer-transfer-function-editor
- https://github.com/InsightSoftwareConsortium/itkwidgets JupyterLab 3 extension

### PR DESCRIPTION
**Pull request by PaulHax**
Some rough types for src/index.js

closes #461

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+declare interface ItkImage {
```
Comment (thewtex) [inline src/index.d.ts:1]: Can we re-use the type definition in the `itk-wasm` package generated from:

 https://github.com/InsightSoftwareConsortium/itk-wasm/blob/master/src/core/Image.ts

?

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  imageType: any
+}
+declare interface ZarrStore {
+  getItem(): any
+}
+declare interface ndarray {
```
Comment (thewtex) [inline src/index.d.ts:7]: @oeway does this look good to you?

File: package.json [hunk 3]
```diff
@@ -124,6 +126,7 @@
     "bundle": "StandaloneHTML ./dist/index.html ./dist/ItkVtkViewer.html",
     "commit": "git cz",
     "format": "prettier --write src/UserInterface/**/*.js src/*.js",
+    "lint:types": "tsc --noEmit",
```
Comment (thewtex) [inline package.json:129]: Can this be part of the `test` npm script?

**Review by thewtex — COMMENTED** [review]
Woohoo!

A few comments inline.

File: src/index.d.ts (hunk 1 above)
Comment (PaulHax) [reply src/index.d.ts]: Yes!

File: src/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+declare interface ItkImage {
+  imageType: any
+}
+declare interface ZarrStore {
```
[the remark this answers is not in the document]
Comment (PaulHax) [reply src/index.d.ts]: Changed to plain old Store now.

File: package.json (hunk 3 above)
Comment (PaulHax) [reply package.json:129]: Done.

**Review by thewtex — APPROVED** [review]
:1st_place_medal: :100: :rocket:

**Pull request merged**